shared/slices/slice_generic.go

```diff
@@ -0,0 +1,308 @@
+package slices
+
+import (
+	"fmt"
+	"reflect"
+)
+
+func interfaceToSlice(slice interface{}) []interface{} {
+	s := reflect.ValueOf(slice)
+	if s.Kind() != reflect.Slice {
+		panic("InterfaceSlice() given a non-slice type")
+	}
+	ret := make([]interface{}, s.Len())
+	for i := 0; i < s.Len(); i++ {
+		ret[i] = s.Index(i).Interface()
+	}
+	return ret
+}
+
+// GenericIntersection returns a new set with elements that are common in
+// both sets a and b.
+func GenericIntersection(a, b interface{}) (reflect.Value, error) {
```

Comment:
Hey @ratikesh9 can we rename these to `Intersection`, `Union`, etc.? Also, would be good to have some simple benchmark tests as reflect is known to be inefficient sometimes. I'd love to see a simple comparison of say, the generic intersection function with reflect and an intersection function for string or uint32 slices without reflect. You can use the `*testing.B` type and its utilities for this in your test file.

Reply:
Sure will add benchmark test for the same

Reply:
@rauljordan  package already contain same function name thats why i changes the name.

Reply:
Benchmark added for the code reflection and non reflection type